From 197507dd6828f14aedc9010813dfb5a77dd02694 Mon Sep 17 00:00:00 2001
From: John Howard <howardjohn@google.com>
Date: Mon, 21 Mar 2022 10:00:06 -0700
Subject: Add util functions for reading directories in tests (#38030)

* Add util functions for reading directories in tests

* use os
---
 pkg/test/util/file/file.go | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/pkg/test/util/file/file.go b/pkg/test/util/file/file.go
index d850c53a51..e1f685f15f 100644
--- a/pkg/test/util/file/file.go
+++ b/pkg/test/util/file/file.go
@@ -128,3 +128,34 @@ func ReadTarFile(filePath string) (string, error) {
 	}
 	return "", fmt.Errorf("file not found %v", filePath)
 }
+
+// ReadDir returns the names of all files in the given directory. This is not recursive.
+// The base path is appended; for example, ReadDir("dir") -> ["dir/file1", "dir/folder1"]
+func ReadDir(filePath string, extensions ...string) ([]string, error) {
+	dir, err := os.ReadDir(filePath)
+	if err != nil {
+		return nil, err
+	}
+	res := []string{}
+	for _, d := range dir {
+		matched := len(extensions) == 0 // If none are set, match anything
+		for _, ext := range extensions {
+			if filepath.Ext(d.Name()) == ext {
+				matched = true
+				break
+			}
+		}
+		if matched {
+			res = append(res, filepath.Join(filePath, d.Name()))
+		}
+	}
+	return res, nil
+}
+
+func ReadDirOrFail(t test.Failer, filePath string, extensions ...string) []string {
+	res, err := ReadDir(filePath, extensions...)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return res
+}
-- 
2.35.3

